src/incidentFeed.ts

```diff
@@ -0,0 +1,128 @@
+import { isToday, isYesterday } from 'date-fns';
+import { Notification, shell } from 'electron'; // tslint:disable-line no-implicit-dependencies
+import settings from 'electron-settings';
+import { EventEmitter } from 'events';
+import Parser from 'rss-parser';
+
+const FEED_URL = 'https://www.netlifystatus.com/history.rss';
+
+interface IFeedItem {
+  title: string;
+  pubDate: string;
+  content: string;
+  link: string;
+}
+
+interface IConnection {
+  isOnline: boolean;
+}
+
+export default class IncidentFeed extends EventEmitter {
+  private parser: { parseURL(feedUrl: string) };
+  private feed: IFeedItem[];
+
+  constructor(connection: IConnection) {
+    super();
+    this.parser = new Parser();
+    this.feed = [];
+    let isInitialFetch = true;
+
+    // main update loop for checking Netlify Incidents RSS
+    const repeat = () => {
+      setTimeout(
```

Comment:
In the long run, I was planning to have a separated scheduler for everything. Do you think it would be alright to have all the "repeat" handling in the same place with a comment like "soon there will be a scheduler".

I'm dreaming of something like 

``` 
scheduler.repeat( () => {
  feed.getFeed()
}, 5000)
```

this would take care of online/offline, too.

Could IncidentFeed expose one function that is called around [here](https://github.com/stefanjudis/netlify-menubar/blob/master/src/menubar.ts#L99-L112)?

In my mind, it feels a little bit that it could be 

```
repeat( () => {
  const incidents = feed.getNewItems()

  if (incidents.filter(isNew)) // notify
})
```

This way the feed class would deal with the fetching and figuring out what's new and notifications and scheduling happen outside.

Reply:
Love the scheduler idea 👍Will move the repeat loop to menubar in the meantime.